http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoException.scala

```diff
@@ -7,8 +7,9 @@ import org.broadinstitute.dsde.workbench.leonardo.http.errorReportSource
 class LeoException(val message: String = null,
                    val statusCode: StatusCode = StatusCodes.InternalServerError,
                    val cause: Throwable = null)
-    extends WorkbenchException(message) {
+    extends WorkbenchException(message, cause) {
```

Comment:
`LeoException` wasn't overriding `WorkbenchException`'s `cause` field which defaults to `null` and hence swallowing information that is important for this PR's purposes, and probably other use cases' as well.